lxc/utils.go

```diff
@@ -109,6 +109,44 @@ func parseDeviceOverrides(deviceOverrideArgs []string) (map[string]map[string]st
 	return deviceMap, nil
 }
 
+// IsAliasesSubset returns true if the first array is completely contained in the second array.
+func IsAliasesSubset(a1 []api.ImageAlias, a2 []api.ImageAlias) bool {
+	set := make(map[string]interface{})
+	for _, alias := range a2 {
+		set[alias.Name] = nil
+	}
+
+	for _, alias := range a1 {
+		_, found := set[alias.Name]
+		if !found {
+			return false
+		}
+	}
+
+	return true
+}
+
+// GetCommonAliases returns the common aliases between a list of aliases and all the existing ones.
+func GetCommonAliases(client lxd.InstanceServer, aliases []api.ImageAlias) ([]api.ImageAliasesEntry, error) {
+	if len(aliases) == 0 {
+		return nil, nil
+	}
+
+	names := make([]string, len(aliases))
+	for i, alias := range aliases {
+		names[i] = alias.Name
+	}
+
+	sort.Strings(names)
```

Comment:
Why do we sort them?

Reply:
`GetExistingAliases` which is using `sort.SearchStrings` requires sorted slice. I've added comment.